stacks-signer/src/v0/signer.rs

```diff
@@ -557,32 +560,37 @@ impl Signer {
         //  the signer needs to be able to determine whether or not the block they're about to sign would conflict with an already-signed Stacks block
         let signer_signature_hash = block_proposal.block.header.signer_signature_hash();
         if let Some(block_info) = self.block_lookup_by_reward_cycle(&signer_signature_hash) {
-            let Some(block_response) = self.determine_response(&block_info) else {
-                // We are still waiting for a response for this block. Do nothing.
-                debug!("{self}: Received a block proposal for a block we are already validating.";
-                    "signer_sighash" => %signer_signature_hash,
-                    "block_id" => %block_proposal.block.block_id()
-                );
-                return;
-            };
-            // Submit a proposal response to the .signers contract for miners
-            debug!("{self}: Broadcasting a block response to stacks node: {block_response:?}");
-            let accepted = matches!(block_response, BlockResponse::Accepted(..));
-            match self
-                .stackerdb
-                .send_message_with_retry::<SignerMessage>(block_response.into())
-            {
-                Ok(_) => {
+            if should_reevaluate_block(&block_info) {
+                // Treat this case the same as if no block info was found
+            } else {
```

Comment:
This control-flow is a little hard to follow, maybe structuring like this:

```rust
let prior_evaluation = self
    .block_lookup_by_reward_cycle(&signer_signature_hash)
    .and_then(|block_info| if should_reevaluate_block(&block_info) {
        debug!("Received a proposal for this block before, but our rejection reason allows us to reconsider"; "reject_reason" => reject_reason); 
        None
    } else {
        Some(block_info)
    });

// we previously considered this proposal, handle the status here
if let Some(block_info) = prior_evaluation {
  return handle_prior_proposal_eval(block_info);
}
...
```

Reply:
Thanks, I like that. I wasn't super happy with that flow either, but didn't come up with anything cleaner.

Reply:
Refactored in 658e051ddf63313fe65b3c395a8a246d274bc4a3.